### PR TITLE
Method to get all text parts and complete text body

### PR DESCRIPTION
**Pull request by arwinvdv**
I tested the library with a mail with 2 attachments send from an iPad. But I didn't get the right body because of multiple plain text parts.

I added the `getCompleteBodyText` function to get all plain text parts as one string.

### CODE REVIEW

File: src/Message/BasicMessageInterface.php [hunk 1]
```diff
@@ -116,23 +116,23 @@ public function getInReplyTo(): array;
...
+    public function getAllContentsByType(string $type): array;
 
     /**
-     * Get all body HTML parts as array.
+     * Get first body HTML part.
      *
-     * @return string[]
+     * @return null|string Null if message has no HTML message part
      */
-    public function getBodyHtmlParts(): array;
```
Comment (Slamdunk) [inline src/Message/BasicMessageInterface.php:130]: Hi, I won't release a new MAJOR just for this fix, so please don't add BC-Breaks

**Review by Slamdunk — CHANGES_REQUESTED** [review]

File: src/Message/BasicMessageInterface.php (hunk 1 above)
Comment (arwinvdv) [reply src/Message/BasicMessageInterface.php]: @Slamdunk thank for the feedback, good point, I restored the function

**Comment by rudiedirkx** [comment]
Shouldn't `getAllContentsByType` be called `getAllContentsBySubtype` since it checks for subtypes. I would use it to find the `DELIVERY-STATUS` AND `RFC822` attachments, which are **subtypes**. Maybe it's too verbose, but more clear IMO.

**Comment by Slamdunk** [comment]
@rudiedirkx can you test this patch to see if it solves https://github.com/ddeboer/imap/issues/561?

**Comment by rudiedirkx** [comment]
> @rudiedirkx can you test this patch to see if it solves #561?

It doesn't. It can't because the problem lies somewhere else, in 'attachment detection' in `AbstractPart::lazyParseStructure()` and/or `AbstractPart::isAttachment()`. It's not about finding the right part. That's already possible with `new RecursiveIteratorIterator($message, RecursiveIteratorIterator::SELF_FIRST)`. That could be made much easier in this PR with a `getAllPartsBy(Sub)type`, but that's more for me than for this PR.

**Comment by petraeus** [comment]
Hello, can you return subtype SUBTYPE_RFC822 along with SUBTYPE_PLAIN in getCompleteBodyText message/delivery-status is a subtype used by gmail to return extended smtp error code

**Pull request merged**